ibis/pandas/execution/tests/test_window.py

```diff
@@ -582,3 +584,124 @@ def count_both(v1, v2):
         t[control].count().over(row_window).execute(),
         check_names=False,
     )
+
+
+def test_custom_window_udf(t, df):
+    """ Test implementing  a (dummy) custom window.
+
+    This test covers the advance use case to support custom window with udfs.
+
+    Note that method used in this example (e.g, get_preceding, get_aggcontext)
+    are unstable developer API, not stable public API.
+    """
+
+    @reduction(input_type=[dt.float64], output_type=dt.float64)
+    def my_sum(v):
+        return v.sum()
+
+    class CustomInterval(object):
```

Comment:
I think Ibis is Python 3 only. If I'm not missing anything, subclassing object is unnecessary for Python 3.
```suggestion
    class CustomInterval:
```